packages/app/src/cli/services/bulk-operations/execute-bulk-operation.ts

```diff
@@ -64,31 +67,45 @@ export async function executeBulkOperation(input: ExecuteBulkOperationInput): Pr
     return
   }
 
-  const result = bulkOperationResponse?.bulkOperation
-  if (result) {
-    const infoSections = [
-      {
-        title: 'Bulk Operation Created',
-        body: [
-          {
-            list: {
-              items: [
-                outputContent`ID: ${outputToken.cyan(result.id)}`.value,
-                outputContent`Status: ${outputToken.yellow(result.status)}`.value,
-                outputContent`Created: ${outputToken.gray(String(result.createdAt))}`.value,
-              ],
-            },
-          },
-        ],
-      },
-    ]
-
-    renderInfo({customSections: infoSections})
-
-    renderSuccess({
-      headline: 'Bulk operation started successfully!',
-      body: 'Congrats!',
-    })
+  const createdOperation = bulkOperationResponse?.bulkOperation
+  if (createdOperation) {
+    if (watch) {
+      const finishedOperation = await watchBulkOperation(adminSession, createdOperation.id)
+      renderBulkOperationResult(finishedOperation)
+    } else {
+      renderBulkOperationResult(createdOperation)
+    }
+  }
```

Comment:
What about simplifying this way?

```suggestion
  let operation = bulkOperationResponse?.bulkOperation
  if (!operation) throw new AbortError('Something went wrong when creating the bulk operation')
  if (watch) operation = await watchBulkOperation(adminSession, createdOperation.id)
  renderBulkOperationResult(operation)
```